core/remote/cozy.js

```diff
@@ -526,13 +529,83 @@ async function getChangesFeed(
   if (pending === 0) {
     return response
   }
-  const nextResponse = await getChangesFeed(last_seq, client)
+  const nextResponse = await fetchChangesFromFeed(last_seq, client)
   return {
     ...nextResponse,
     results: [...results, ...nextResponse.results]
   }
 }
 
+async function fetchInitialChanges(
+  client /*: OldCozyClient */
+) /*: Promise<ChangesFeedResponse> */ {
+  const last_seq = await fetchRemoteLastSequence(client)
+  const results = await fetchAllDocs(client)
+
+  return { pending: 0, last_seq, results }
+}
+
+async function fetchRemoteLastSequence(client /*: OldCozyClient */) {
+  const response = await client.data.changesFeed(FILES_DOCTYPE, {
+    limit: 1,
+    descending: true
+  })
+  return response.last_seq
+}
+
+async function fetchAllDocs(
+  client /*: OldCozyClient */,
+  batchSize /*: number */ = 10000
+) /*: Promise<Array<{ doc: RemoteDoc|RemoteDeletion }>> */ {
+  let hasMore = true
+  let startDocId = null
+  let docs /*: RemoteDoc[] */ = []
+
+  while (hasMore) {
+    if (!startDocId) {
+      // No startDocId set: this is the first time we replicate
+      const rows = await client.data.findAll(FILES_DOCTYPE, {
+        limit: batchSize,
+        include_docs: true
+      })
+      docs = filterDocs(rows)
+      if (docs.length === 0) {
+        hasMore = false
+      } else {
+        startDocId = docs[docs.length - 1]._id
+        if (docs.length < batchSize) {
+          hasMore = false
+        }
+      }
+    } else {
+      const rows = await client.data.findAll(FILES_DOCTYPE, {
+        include_docs: true,
+        limit: batchSize,
+        startkey_docid: startDocId
+      })
+      const filteredDocs = filterDocs(rows)
+      // FIXME: why less than 2??? Use descriptive constant instead.
+      if (filteredDocs.length < 2) {
+        return docs.map(d => ({ doc: d }))
```

Comment:
It's kind of a pity having to iterate over 10K docs just add the `doc` key :/ Is it really necessary? If yes, it should be more efficient to do it when we filter the docs after the fetch, so we only iterate once?

Reply:
Note we also iterate over all the docs in [cozy-client-js](https://github.com/cozy/cozy-client-js/blob/4daaf6bb186cf17a51c672fca287b7aa63997880/src/data.js#L107), so it might be even better to fetch manually, just like [here](https://github.com/cozy/cozy-client/blob/812c763048b32568947e6e6c96757bea4409ef05/packages/cozy-pouch-link/src/remote.js#L10)